crates/wasi-common/src/sandboxed_tty_writer.rs

```diff
@@ -0,0 +1,177 @@
+use std::io::{Result, Write};
+
+/// An adapter around a `Write` stream that guarantees that its output
+/// is valid UTF-8 and contains no control characters. It does this by
+/// replacing characters with inert control pictures and replacement
+/// characters.
+pub(crate) struct SandboxedTTYWriter<'writer, Writer>
+where
+    Writer: Write,
+{
+    inner: &'writer mut Writer,
+    scratch: [u8; 4],
```

Comment:
I reckon it'd be good to add a comment here that the buffer of 4 bytes is large enough to encode any char, as per the `libstd` docs [primitive.char.html#method.encode_utf8](https://doc.rust-lang.org/std/primitive.char.html#method.encode_utf8).

Reply:
Done!